src/App.tsx

```diff
@@ -1,55 +1,79 @@
-import React from 'react';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import debounce from 'lodash/debounce';
+import cn from 'classnames';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { Autocomplete } from './components/Autocompete';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPersonSlug, setSelectedPersonSlug] = useState('');
+  const [query, setQuery] = useState('');
+  const [debouncedQuery, setDebouncedQuery] = useState('');
+  const [isTyping, setIsTyping] = useState(false);
+  const selectedPerson: Person | undefined = peopleFromServer
+    .find(person => person.slug === selectedPersonSlug);
+
+  const applyQuery = useCallback(
+    debounce(setDebouncedQuery, 500),
+    [],
+  );
+
+  const onSelect = useCallback((slug: string) => {
+    setSelectedPersonSlug(slug);
+  }, [debouncedQuery]);
+
+  const applyTypingDelay = useCallback(
+    debounce(setIsTyping, 500),
+    [],
+  );
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    setIsTyping(true);
+    applyQuery(event.target.value);
+    applyTypingDelay(false);
+  };
+
+  useEffect(() => {
+    setQuery('');
+    setDebouncedQuery('');
+    setIsTyping(false);
+  }, [selectedPerson]);
+
+  const dropDownList = useMemo(() => {
+    return peopleFromServer.filter(person => (
+      person.name.toLowerCase().includes(debouncedQuery.toLowerCase())
+    ));
+  }, [debouncedQuery]);
 
   return (
     <main className="section">
       <h1 className="title">
-        {`${name} (${born} = ${died})`}
+        {
+          selectedPerson
+            ? `${selectedPerson?.name} (${selectedPerson?.born} = ${selectedPerson?.died})`
+            : 'No selected person'
+        }
```

Comment:
looks more ergonomic this way
```suggestion
        {selectedPerson
            ? `${selectedPerson?.name} (${selectedPerson?.born} = ${selectedPerson?.died})`
            : 'No selected person'}
```